client/state/current-user/reducer.js

```diff
@@ -60,28 +61,35 @@ export const currencyCode = createReducer( null, {
  * @param  {Object} action Action payload
  * @return {Object}        Updated state
  */
-export const capabilities = createReducer( {}, {
-	[ SITE_RECEIVE ]: ( state, action ) => {
-		if ( ! action.site.capabilities ) {
-			return state;
-		}
+export const capabilities = withSchemaValidation(
+	capabilitiesSchema,
+	( state = {}, action ) => {
```

Comment:
I thought I had already left a comment here but GitHub isn't showing it to me, so please pardon me if I'm redundant here (it was from my phone, so maybe that's why)

it's rare that I think it's a good idea to define a function inside of the argument list of another function call. in fact, I usually consider it a bad idea (such as inlining the mapping functions inside of `connect()`).

if we pull this out to its own named function (and export) then it's easier to test in isolation and I think it's also easier to see what's going on, that we have a raw reducer and then we have a composed version of it which adds schema validation.

last time I saw the discussion, you and @gwwar were both in favor of inlining inside of `connect()`, although I think we can agree that we're all using a subjective line there or else we would also define our React components inside of the call to `connect()`.

in this case, I feel the line has been crossed again just because we can't test the reducer in isolation or recompose it in other ways.

hope this wasn't too rantful :smile: